### PR TITLE
Rename IntList to IntArrayRef.

### PR DESCRIPTION
**Pull request by ezyang**
Stack:
&nbsp;&nbsp;&nbsp;&nbsp;:black_circle:&nbsp; **#16751 Rename IntList to IntArrayRef.**&nbsp;&nbsp;[:green_heart:](https://our.internmc.facebook.com/intern/diff/D13954363/)

This was made more complicated by the fact that ivalue::IntList
is a thing.  So I had to fix all of the sites where we referring
to IValue post facto.

The following codemods were run, in this order:

```
codemod -m -d . --extensions cc,cpp,cu,cuh,h,hpp,py,cwrap,yaml,in IntList IntArrayRef
codemod -m -d . --extensions cc,cpp,cu,cuh,h,hpp,py,cwrap,yaml,in IntArrayRef::create IntList::create
codemod -m -d . --extensions cc,cpp,cu,cuh,h,hpp,py,cwrap,yaml,in ivalue::IntArrayRef ivalue::IntList
codemod -m -d . --extensions cc,cpp,cu,cuh,h,hpp,py,cwrap,yaml,in Tag::IntArrayRef Tag::IntList
codemod -m -d . --extensions cc,cpp,cu,cuh,h,hpp,py,cwrap,yaml,in isIntArrayRef isIntList
codemod -m -d . --extensions cc,cpp,cu,cuh,h,hpp,py,cwrap,yaml,in toIntArrayRef toIntList
codemod -m -d . --extensions cc,cpp,cu,cuh,h,hpp,py,cwrap,yaml,in 'Shared<IntArrayRef>' 'Shared<IntList>'
codemod -m -d . --extensions cc,cpp,cu,cuh,h,hpp,py,cwrap,yaml,in 'intrusive_ptr<IntArrayRef>' 'intrusive_ptr<IntList>'
```

Differential Revision: [D13954363](https://our.internmc.facebook.com/intern/diff/D13954363/)

### CODE REVIEW

File: aten/src/ATen/core/ivalue.h [hunk 1]
```diff
@@ -135,7 +135,7 @@ using GenericDict = Dict<IValue, IValue>;
   _(Int) \
   _(Bool) \
   _(Tuple) \
-  _(IntList) \
+  _(IntArrayRef) \
```
Comment (gchanan) [inline aten/src/ATen/core/ivalue.h]: this looks wrong.

File: aten/src/ATen/core/ivalue.h [hunk 2]
```diff
@@ -319,7 +319,7 @@ struct CAFFE2_API IValue final {
     return payload.as_bool;
   }
 
-  // IntList
+  // IntArrayRef
```
Comment (gchanan) [inline aten/src/ATen/core/ivalue.h]: here too.

**Comment by ezyang** [comment]
@gchanan Ya, those got fixed in the manual fixes PR (stacked on top). I'll squash at the end; they're separate for now for ease of review.

**Comment by gchanan** [comment]
I guess I'd quibble with making it easier to review...seeing tests pass and no BC change (to be sure you got all the places) requires looking at some combination of stacked and partial number of commits within a PR?

**Comment by ezyang** [comment]
I'm perfectly happy to squash the PRs now; after all, you are the reviewer :)  I split because I remember instances when I was reviewing Jerry's PRs and I didn't know which places to focus in on for human mistakes, because they were interspersed with automated changes. But I guess a review of all of the codemodded code is still in order, so maybe it is easier if they have been integrated
 
Re tests, we only care about https://github.com/pytorch/pytorch/pull/16752

**Comment by ezyang** [comment]
Ignoring clang-tidy error; I'll fix it in follow up

**Pull request closed without merging**